### DISCUSSION TITLE
Private Github Data Source

### DISCUSSION THREAD

**Discussion by adamhemsley** [netbox-community/netbox#16370] [Q&A] [answered]
Hi Guys,

Does anyone know if there is any way to use a private github repo as a data source for custom scripts / config templates?
If I make my repo private netbox can no longer sync (presumably because the repos url no longer resolves). Giving a username / password doesn't seem to help.

Any advise appreciated.

Adam

**Comment by Mailstorm-ctrl** [comment, marked as answer]
See how I did it here:
https://github.com/netbox-community/netbox/issues/15394

**Reply by adamhemsley** [reply]
Legend, many thanks.

**Comment by DC-Rasmus-Elmholt** [comment]
Any suggestions on how this is handled when Netbox is running in docker?
